android-components/components/browser/engine-gecko/src/main/java/mozilla/components/experiment/NimbusExperimentDelegate.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.experiment
+
+import mozilla.components.browser.engine.gecko.GeckoNimbus
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONObject
+import org.mozilla.geckoview.ExperimentDelegate
+import org.mozilla.geckoview.GeckoResult
+
+/**
+ * Default Nimbus [ExperimentDelegate] implementation to communicate with mobile Gecko and GeckoView.
+ */
+class NimbusExperimentDelegate : ExperimentDelegate {
+
+    private val logger = Logger(NimbusExperimentDelegate::javaClass.name)
+
+    /**
+     * Retrieves experiment information on the feature for use in GeckoView.
+     *
+     * @param feature Nimbus feature to retrieve information about
+     * @return a JSON object containing experiment information
+     */
+    override fun onGetExperimentFeature(feature: String): GeckoResult<JSONObject>? {
+        val result = GeckoResult<JSONObject>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
+        if (nimbusFeature != null) {
+            result.complete(nimbusFeature.toJSONObject())
+        } else {
+            logger.warn("Could not find Nimbus feature '$feature' to retrieve experiment information.")
+            result.complete(null)
+        }
+        return result
+    }
+
+    /**
+     * Records that an exposure event occurred with the feature.
+     *
+     * @param feature Nimbus feature to record information about
+     * @return a [GeckoResult] with true if the feature was found and recorded, false if not found
+     */
+    override fun onRecordExposureEvent(feature: String): GeckoResult<Boolean>? {
+        val result = GeckoResult<Boolean>()
+        val nimbusFeature = GeckoNimbus.getFeature(feature)
+        if (nimbusFeature != null) {
+            nimbusFeature.recordExposure()
+            result.complete(true)
+        } else {
+            logger.warn("Could not find Nimbus feature '$feature' to record an exposure event.")
+            result.complete(false)
+        }
+        return result
+    }
```

Comment:
At some point— perhaps in this PR— I'd like to see support for the other `record*` methods in `FeatureHolder`, specifically the [`recordExperimentExposure` method](https://github.com/mozilla/application-services/blob/main/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/internal/FeatureHolder.kt#L74) ([docs](https://experimenter.info/fml/coenrolling-features#recording-exposure)) and [the `recordMalformedConfiguration`](https://github.com/mozilla/application-services/blob/main/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/internal/FeatureHolder.kt#L91).

Reply:
Thanks, glad you mentioned that! Adjusted to expand and cover those methods

Reply:
So good! Thank you!